### PR TITLE
Add CPPM VM cmdlets

### PR DESCRIPTION
**Pull request by alagoutte**
For autodeploy and initial configure of CPPM VM

### CODE REVIEW

**Comment by alagoutte** [comment]
@CedricMoreau can you look ? (WIP)

File: PowerArubaCP/Public/VM.ps1 [hunk 1]
```diff
@@ -0,0 +1,751 @@
...
+        [string]$hostname,
+        [Parameter (Mandatory = $true)]
+        [ipaddress]$mgmt_ip,
+        [Parameter (Mandatory = $true)]
+        [ValidateRange(0, 32)]
+        [int]$mgmt_netmask,
+        [Parameter (Mandatory = $true)]
+        [ipaddress]$mgmt_gateway,
+        [Parameter (Mandatory = $false)]
+        [ipaddress]$data_ip,
```
Comment (CedricMoreau) [inline PowerArubaCP/Public/VM.ps1:326]: Is data port really mandatory ?

File: PowerArubaCP/Public/VM.ps1 [hunk 2]
```diff
@@ -0,0 +1,751 @@
...
+        [Parameter (Mandatory = $false)]
+        [switch]$fips
+    )
+
+    Begin {
+    }
+
+    Process {
+        if ($timezone_continent -xor $timezone_country) {
+            Throw "You need to specific Timezone Continent and Country on the sametime"
```
Comment (CedricMoreau) [inline PowerArubaCP/Public/VM.ps1:356]: same time

File: PowerArubaCP/Public/VM.ps1 [hunk 3]
```diff
@@ -0,0 +1,751 @@
...
+
+            #Management IPv6 (Skip...)
+            Write-Output "Skip Configure Management IPv6..."
+            Set-VMKeystrokes -VMName $name_vm -SpecialKeyInput "KeyEnter" 6>> $null
+            Start-Sleep 1
+        }
+
+        #Data Port
+        if ($version -eq "6.8") {
+            if ($data_ip -and $data_netmask -and $data_gateway) {
```
Comment (CedricMoreau) [inline PowerArubaCP/Public/VM.ps1:412]: Data configuration is mandatory but here you check if they have values, is is usefull ?

File: PowerArubaCP/Public/VM.ps1 [hunk 4]
```diff
@@ -0,0 +1,751 @@
...
+
+            #SSH Password
+            Write-Output "Set SSH Password"
+            Set-VMKeystrokes -VMName $name_vm -StringInput $ssh_password -ReturnCarriage $true 6>> $null
+            Start-Sleep 1
+        }
+    }
+
+    End {
+    }
```
Comment (CedricMoreau) [inline PowerArubaCP/Public/VM.ps1:750]: Blank line ?

File: PowerArubaCP/Public/VM.ps1 (hunk 1 above)
Comment (alagoutte) [reply PowerArubaCP/Public/VM.ps1:326]: it is not mandatory

File: PowerArubaCP/Public/VM.ps1 (hunk 2 above)
Comment (alagoutte) [reply PowerArubaCP/Public/VM.ps1:356]: You need timezone continent and country (too complicated to use ParameterSetName on this case)

**Pull request merged**